presto-main/src/main/java/com/facebook/presto/server/remotetask/TaskInfoFetcher.java

```diff
@@ -209,9 +215,14 @@ private synchronized void sendNextRequest()
 
         HttpUriBuilder httpUriBuilder = uriBuilderFrom(taskStatus.getSelf());
         URI uri = summarizeTaskInfo ? httpUriBuilder.addParameter("summarize").build() : httpUriBuilder.build();
-        Request request = setContentTypeHeaders(isBinaryTransportEnabled, prepareGet())
-                .setUri(uri)
-                .build();
+        Request.Builder uriBuilder = setContentTypeHeaders(isBinaryTransportEnabled, prepareGet());
+
+        if (taskInfoRefreshMaxWait.toMillis() > 0) {
+            uriBuilder.setHeader(PRESTO_CURRENT_STATE, taskStatus.getState().toString())
```

Comment:
So if task status get changed, worker will immediately response; otherwise, it will wait up to `taskInfoRefreshMaxWait` ?

Reply:
yes! Correct.